### DISCUSSION TITLE
Backup data from faulty disk with Borg

### DISCUSSION THREAD

**Discussion by beta2k** [borgbackup/borg#8339] [General]
My hosting company allows the possibility to backup data with Borg. I have a faulty disc and want to backup as much data as possible from this faulty disc. Therefore I want to ask, if this is possible with Borg? Do I need to set any special parameters when using `borg create` when I know that there will be bad sectors? Or will these just be skipped by borg? Obviously I don't want Borg to backup faulty data and then restore this faulty data onto the new working HDD.

I scanned through the parameters of the `borg create` command and didn't find anything obvious pointing towards special behaviour when dealing with a faulty disk

Appreciate your replies.

**Comment by ThomasWaldmann** [comment]
If borg create gets an IOError from the OS for some file, it will report that file with "E" (when using `--list`), skip it and proceed with the next file. If borg determines a file to be unchanged, it does not even read it, but uses chunks that already exist in the repo and information from the files cache.

Depending on how many retries the OS does before reporting an error, how long it takes per retry and how many IO errors there are, it might be slow.

I'ld keep the list from the log output, so it can be reviewed later.
